ui/src/app/components/gke-app-configuration-panel.spec.tsx

```diff
@@ -413,84 +378,131 @@ describe(GKEAppConfigurationPanel.name, () => {
     jest
       .spyOn(disksApi(), 'listOwnedDisksInWorkspace')
       .mockImplementation((): Promise<any> => Promise.resolve([disk]));
-    const wrapper = createWrapper({
+    createWrapper({
       onClose: onCloseStub,
-      type: AppType.CROMWELL,
+      appType: AppType.CROMWELL,
       workspaceNamespace,
     });
-    await waitOneTickAndUpdate(wrapper);
-    // Start with the DELETE_UNATTACHED_PD panel
-    act(() => {
-      wrapper
-        .find(CromwellConfigurationPanel)
-        .prop('onClickDeleteUnattachedPersistentDisk')();
+
+    await waitFor(() => {
+      expect(screen.queryByText(cromwellIntroTextRegex)).not.toBeNull();
     });
-    await waitOneTickAndUpdate(wrapper);
 
-    wrapper.find(ConfirmDeleteUnattachedPD).prop('onConfirm')();
-    await waitOneTickAndUpdate(wrapper);
+    const deleteButton = screen.getByRole('button', {
+      name: 'Delete Persistent Disk',
+    });
+    expectButtonElementEnabled(deleteButton);
+    deleteButton.click();
 
-    expect(deleteUnattachedPDStub).toHaveBeenCalledWith(
-      workspaceNamespace,
-      disk.name
-    );
-    expect(onCloseStub).toHaveBeenCalled();
+    await waitFor(() => {
+      expect(screen.queryByText(deleteUnattachedPdRegex)).not.toBeNull();
+    });
 
-    // Validate there is no error modal if the call succeeds
-    expect(notificationStore.get()).toBeNull();
+    const deletePDRadioButton = screen.getByRole('radio', {
+      name: 'Delete persistent disk',
+    });
+    deletePDRadioButton.click();
+
+    const confirmDeleteButton = screen.getByRole('button', { name: 'Delete' });
+    expectButtonElementEnabled(confirmDeleteButton);
+    confirmDeleteButton.click();
+
+    await waitFor(() => {
+      expect(deleteUnattachedPDStub).toHaveBeenCalledWith(
+        workspaceNamespace,
+        disk.name
+      );
+      expect(onCloseStub).toHaveBeenCalled();
+
+      // Validate there is no error modal if the call succeeds
+      expect(notificationStore.get()).toBeNull();
+    });
   });
 
   it('should show an error modal if the delete PD API call fails', async () => {
     jest
       .spyOn(disksApi(), 'deleteDisk')
       .mockImplementation((): Promise<any> => Promise.reject());
 
-    const wrapper = createWrapper({
-      type: AppType.CROMWELL,
+    const disk = stubDisk();
```

Comment:
need a PD for this test - I guess the previous test's `props` manipulation didn't need this